### DISCUSSION TITLE
What is Redis used for?

### DISCUSSION THREAD

**Discussion by RainerZufall187** [mastodon/mastodon#22737] [Q&A] [answered]
ok

**Comment by rbairwell** [comment, marked as answer]
Caching of timelines and other volatile items. If it's read from the database, it's most likely to be stored in Redis for a short period of time to reduce database accesses. See https://docs.joinmastodon.org/admin/scaling/#redis and https://docs.joinmastodon.org/admin/config/#redis

**Reply by RainerZufall187** [reply]
@rbairwell
> Caching of timelines and other volatile items. If it's read from the database, it's most likely to be stored in Redis for a short period of time to reduce database accesses. See https://docs.joinmastodon.org/admin/scaling/#redis and https://docs.joinmastodon.org/admin/config/#redis

Do I need to back it up?

**Reply by ClearlyClaire** [reply]
It probably should not be backed up because of how time-sensitive the data is. However, Redis is not just used for caching, it's used to queue all kinds of background jobs. Therefore, you should avoid clearing it, and you should backup/transfer it if you ever need to transfer your server's setup from one machine to another.

**Comment by mehran-prs** [comment]
Is it used to cache status models themselves (not just their IDs in feed lists)?

**Reply by ClearlyClaire** [reply]
Mastodon actually uses up to 3 different Redis instances for different purposes:
- Rails cache (somewhat short-lived cache): this may contain some status attributes (e.g. when translating, or rendered status JSON that is about to be delivered to multiple recipients) but as of 4.3.0, it should not contain status models themselves
- application data: this is for volatile data and synchronization, it should not contain status models; this contains the feeds, which are basically just IDs with a bit of metadata, but neither the status models nor their contents
- Sidekiq: this is for enqueuing jobs and synchronizing background tasks, this may contain rendered JSON of to-be-sent or to-be-processed statuses but not the models themselves

**Reply by mehran-prs** [reply]
Thanks @ClearlyClaire , 
So just to make sure, when I fetch my home timeline, it just fetches my home timeline status IDs from Redis and then fetches all statues from DB and renders them(Rails may cache rendered version here), is this true?

**Reply by ClearlyClaire** [reply]
If you're using Mastodon 4.3.0 or newer, yes, IDs are fetched from Redis, and data is fetched from the database. Rendered statuses are also not cached in this path.
